PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelperData.cxx

```diff
@@ -0,0 +1,199 @@
+      }
+    }
+  }
+
+  TString histName = "fHistPtSelection";
+  if(nTracksInRange==0) {
+    fHistManager.FillTH2(histName, 1.5, ptMax );
+    return kFALSE;
+  }
+  else {
+    fHistManager.FillTH2(histName, 0.5, ptMax );
+  }
+
+
+  return kTRUE;
+}
+
+
+void AliAnalysisTaskEmcalEmbeddingHelperData::RetrieveTaskPropertiesFromYAMLConfig() {
+
+  AliAnalysisTaskEmcalEmbeddingHelper::RetrieveTaskPropertiesFromYAMLConfig();
+
+  // get pt selection from YAML file
+  // defined in 2 arrays, pt
+
+  std::vector <double> ptSelectionMin;
+  std::vector <double> ptSelectionMax;
+  bool resMin = fYAMLConfig.GetProperty( "ptSelectionMin", ptSelectionMin, false);
+  bool resMax = fYAMLConfig.GetProperty( "ptSelectionMax", ptSelectionMax, false);
+  if (resMin && resMax) {
+    if (ptSelectionMin.size() != ptSelectionMax.size()) {
+      AliErrorStream() << "Passed pt selection min with size" << ptSelectionMin.size() << " entries and pt selection max with size " << ptSelectionMax.size() << " - need same size arrays\n";
+    }
+    else {
+      for(int i = 0; i < ptSelectionMin.size(); i++) {
+        AliDebugStream(1) << "Setting pt selection [" << ptSelectionMin.at(i) << ", " << ptSelectionMax.at(i) << "]\n";
+        fPtSelection.push_back(std::make_pair(ptSelectionMin.at(i), ptSelectionMax.at(i)));
+      }
+    }
+  }
+  else if( (resMin && !resMax) || (!resMin && resMax)) {
+    AliErrorStream() << "Only set one pt selection array - should set both\n";
+  }
+}
+
+/**
+ *
+ * The main implementation of the AddTask. 
+ * Same structure as AddTaskEmbeddingHelper
+ *
+ */
+
+AliAnalysisTaskEmcalEmbeddingHelperData *AliAnalysisTaskEmcalEmbeddingHelperData::AddTaskEmcalEmbeddingHelperData()
+{  
+
+  // Get the pointer to the existing analysis manager via the static access method.
+  //==============================================================================
+  AliAnalysisManager *mgr = AliAnalysisManager::GetAnalysisManager();
+  if (!mgr)
+  {
+    ::Error("AddTaskEmcalEmbeddingHelperData", "No analysis manager to connect to.");
+    return 0;
+  }
+
+  // Check the analysis type using the event handlers connected to the analysis manager.
+  //==============================================================================
+  AliVEventHandler* handler = mgr->GetInputEventHandler();
+  if (!handler)
+  {
+    ::Error("AddTaskEmcalEmbeddingHelperData", "This task requires an input event handler");
+    return 0;
+  }
+
+  TString name = "AliAnalysisTaskEmcalEmbeddingHelperData";
+
+  AliAnalysisTaskEmcalEmbeddingHelperData * mgrTask = static_cast<AliAnalysisTaskEmcalEmbeddingHelperData *>(mgr->GetTask(name.Data()));
+  if (mgrTask) return mgrTask;
+
+  // Create the task that manages
+  AliAnalysisTaskEmcalEmbeddingHelperData * embeddingHelper = new AliAnalysisTaskEmcalEmbeddingHelperData(name.Data());
+
+  // set embedded track container
+  AliTrackContainer *contEmb = new AliTrackContainer("tracks");
+  contEmb->SetTrackCutsPeriod("LHC17p");
```

Comment:
I wouldn't recommend leaving this as the default, but it's of course fine for now. In my opinion, I think it would be best to set it via YAML.